doc/sphinx/proof-engine/ltac.rst

```diff
@@ -532,42 +635,47 @@ tactic that succeeds from a list of tactics:
 
             Tactic Notation "foo" tactic_list(tacs) := first tacs.
 
-Left-biased branching: ||
-~~~~~~~~~~~~~~~~~~~~~~~~~
+First tactic to make progress: ||
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 Yet another way of branching without backtracking is the following
 structure:
 
-.. tacn:: @ltac_expr__1 || @ltac_expr__2
-   :name: || (left-biased branching)
+.. tacn:: @ltac_expr1 || {| @ltac_expr2 | @binder_tactic }
+   :name: || (first tactic making progress)
 
-   :n:`@ltac_expr__1` and :n:`@ltac_expr__2` are evaluated to :n:`v__1` and
-   :n:`v__2` respectively, which must be tactic values. The tactic value :n:`v__1` is
-   applied in each subgoal independently and if it fails *to progress* then
-   :n:`v__2` is applied. :n:`@ltac_expr__1 || @ltac_expr__2` is
-   equivalent to :n:`first [ progress @ltac_expr__1 | @ltac_expr__2 ]` (except that
-   if it fails, it fails like :n:`v__2`). Branching is left-associative.
+   :n:`@ltac_expr1 || @ltac_expr2` is
+   equivalent to :n:`first [ progress @ltac_expr1 | @ltac_expr2 ]`, except that
+   if it fails, it fails like :n:`v__2`. Branching is left-associative.
 
 Generalized biased branching: tryif
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-.. tacn:: tryif @ltac_expr__if then @ltac_expr__then else @ltac_expr2__else
+.. tacn:: tryif @ltac_expr__test then @ltac_expr__then else @ltac_expr2__else
    :name: tryif
 
    is a generalization of the biased-branching tactics above. The
-   expression :n:`@ltac_expr__if` is evaluated to :n:`v__if`, which is then
+   expression :n:`@ltac_expr__test` is evaluated to :n:`v__if`, which is then
    applied to each subgoal independently. For each goal where :n:`v__if`
```

Comment:
The evaluation happens in the context of each subgoal independently.  This can be seen by noting that
```coq
Goal True /\ True.
  split.
  all: tryif (let n := numgoals in idtac n) then idtac else idtac.
```
prints `1` twice unlike
```coq
Goal True /\ True.
  split.
  all: let n := numgoals in [ > idtac n .. ].
```
which prints `2` twice.

Reply:
How is this as a replacement for that paragraph?

   For each focused goal, independently: Evaluates and applies :n:`@ltac_expr__test`.
   If :n:`@ltac_expr__test` succeeds at least once, evaluate and apply :n:`@ltac_expr__then`
   to all the subgoals generated by :n:`@ltac_expr__test`.  Otherwise, evaluate and apply
   :n:`@ltac_expr2__else` to all the subgoals generated by :n:`@ltac_expr__test`.

Reply:
Looks good, except that you use "evaluates" in the first sentence, but use "evaluate" in the other sentences.  The tense should agree between all three sentences.

Reply:
Good point.  To be precise, they are both present tense, but for different numbers (It evaluates vs they evaluate).